rpc/backend/src/test/scala/io/udash/rpc/internals/ExposesServerRPCTest.scala

```diff
@@ -98,10 +100,35 @@ class ExposesServerRPCTest extends UdashBackendTest {
     val impl = TestRPC.rpcImpl((method: String, args: List[List[Any]], result: Option[Any]) => {
       calls += method
     })
-    val rpc = new UPickleExposesServerRPC[TestRPC](impl)
-    rpc
+    new UPickleExposesServerRPC[TestRPC](impl)
+  }
+
+  val loggedCalls = ListBuffer.empty[String]
+
+  def createLoggingRpc(calls: mutable.Builder[String, Seq[String]]): ExposesServerRPC[TestRPC] = {
+    val impl = TestRPC.rpcImpl((method: String, args: List[List[Any]], result: Option[Any]) => {
+      calls += method
+    })
+    new ExposesServerRPC[TestRPC](impl) with CallLogging[TestRPC] {
+      override protected val metadata: RPCMetadata[TestRPC] = RPCMetadata[TestRPC]
+
+      override val framework = DefaultUdashRPCFramework
+
+      import framework._
+
+      override protected def localRpcAsRaw: framework.AsRawRPC[TestRPC] = framework.AsRawRPC[TestRPC]
+
+      override def log(rpcName: String, methodName: String, args: Seq[String]): Unit = loggedCalls += s"$rpcName $methodName $args"
+    }
   }
 
   "DefaultExposesServerRPC" should tests(createDefaultRpc)
   "CustomExposesServerRPC" should tests(createCustomRpc)
+  "LoggingExposesServerRPC" should {
+    tests(createLoggingRpc)
```

Comment:
Usage of tests() in this case is not a good idea, I think it would be better to create separate calls for annotated and unannotated methods here.